GridExtensionPack/src/main/java/org/vaadin/teemusa/gridextensions/paging/PagedDataProvider.java

```diff
@@ -37,10 +39,11 @@ public boolean isInMemory() {
 
 	@Override
 	public int size(Query<T, F> query) {
-		return getPagingControls().getSizeOfPage(query);
```

Comment:
The idea is that PagedDataProvider limits the amount of calls to dataProvider.size which can be a costly operation on some backends. The "size" of the virtual data provider is (0, PAGELENGTH] and will always be in that range. The contents of the paged "virtual" data provider will change, but the size almost always the same. only the last page might be smaller than PAGELENGTH depending on how many items there are.

There might be some issue with the Filtering as is, since the Filter might not be taken into account here, the paged data provider should track the previously used filter and re-use the size if filter stays the same.

But the main point here was to avoid calling the DataProvider#size of the actual data provider when not necessary, and now it will be called every time, which might be a problem for some backends.